doc/ref/chap-builtin-cmd.md

```diff
@@ -1186,3 +1186,60 @@ Flags:
 
 Bash has this, but OSH won't implement it.
 
+
+## Flags
+
+### parseArgs()
+
+YSH includes a `argparse`-like utility called `parseArgs`. This is intended to
+be used for command-line interfaces to programs.
+
+To use it, first import `args.ysh`:
+
+    source --builtin args.ysh
+
+Then, create an argument "spec":
+
+    arg-parse (&spec) {
+      flag -v --verbose (help="Verbosely")  # default is Bool, false
+
+      flag -P --max-procs ('int', default=-1, help='''
+        Run at most P processes at a time
+        ''')
+
+      flag -i --invert ('bool', default=true, help='''
+        Long multiline
+        Description
+        ''')
+
+      arg src (help='Source')
+      arg dest (help='Dest')
+
+      rest files
+    }
+
+Finally, parse `ARGV` (or any other array of strings) with:
+
+    var opt, i = parseArgs(spec, ARGV)
```

Comment:
'opt' is an OK name, but can also be confused with optimization or something

I kind of like the idea that you "parseArgs" and you get "flags"

So then  flag.verbose rather than opt.verbose, because it's concrete.  It's not strictly accurate because you also have flag.src, which is not a flag, but somehow I think that is not horrible

---

Again a separate conversation, although I guess we are documenting it now, so maybe now is the time to flesh out the names

Reply:
For my two cents, I find 'opt' ok, not much to confuse in relation to parsing args, or is it?

I find `flag.something` that isn't a flag is much worse (it's concretely wrong).

If you really think opt could be confused with optimization here, maybe the full word option could also work.